### PR TITLE
Futurize streamlines math

### PR DESCRIPTION
**Pull request by frheault**
Transfert from bitbucket
Tested with Python3

### CODE REVIEW

File: scilpy/utils/streamlines.py [hunk 1]
```diff
@@ -1,9 +1,117 @@
...
+
+def subtraction(left, right):
+    """Subtraction of two streamlines dict (see hash_streamlines)"""
+    return {k: v for k, v in left.items() if k not in right}
+
+
+def union(left, right):
+    """Union of two streamlines dict (see hash_streamlines)"""
+
+    # In python 3 : return {**left, **right}
```
Comment (jchoude) [inline scilpy/utils/streamlines.py]: If we decide to just support Python 3, we could use that syntax

**Comment by jchoude** [comment]
Apart from @GuillaumeTh comments this is good to go for me.

File: scripts/scil_streamlines_math.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    p.add_argument('--save_metadata_indices', '-m', action='store_true',
+                   help='Save streamline indices to metadata. Has no '
+                   'effect if --no-data\nis present. Will '
+                   'overwrite \'ids\' metadata if already present.')
+
+    p.add_argument('--save_indices', '-s', metavar='OUTPUT_INDEX_FILE',
+                   help='Save the streamline indices to the supplied '
+                   'json file.')
+
+    p.add_argument('--verbose', '-v', action='store_true', dest='verbose',
```
Comment (GuillaumeTh) [inline scripts/scil_streamlines_math.py]: Please use function

File: scripts/scil_streamlines_math.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    args = p.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    assert_inputs_exist(p, args.inputs)
+    assert_outputs_exists(p, args, [args.output])
+
+    # Load all input streamlines.
+    data = [load_data(f) for f in args.inputs]
```
Comment (GuillaumeTh) [inline scripts/scil_streamlines_math.py]: Please update with dipy class

**Pull request merged**